### PR TITLE
Use external Babel helpers

### PR DESCRIPTION
**Pull request by Andarist**

### CODE REVIEW

File: .babelrc.js [hunk 1]
```diff
@@ -7,8 +7,22 @@ const presets = [
 ]
 const plugins = ['@babel/plugin-proposal-class-properties']
 
+const transformRuntimeOptions = {
+  version: require('./package.json').dependencies['@babel/runtime'].replace(
+    /^\^/,
+    ''
+  ),
+  regenerator: false,
```
Comment (nicolo-ribaudo) [inline .babelrc.js:15]: With recent `@babel/runtime` versions there is no special handling of regeneratorRuntime anymore, it should be safe to remove this.

Comment (Andarist) [reply .babelrc.js:15]: @nicolo-ribaudo I've set this up based on the docs [here](https://babeljs.io/docs/babel-plugin-transform-runtime#regenerator). What does `regenerator: true/false` means nowadays and what's the default? Are those docs up to date? 🤔

Comment (nicolo-ribaudo) [reply .babelrc.js:15]: Uh yeah... the docs need to be updated. Whether generator functions are transformed using regenerator or not doesn't depend on this option, but by whether or not `@babel/plugin-transform-regenerator` is enabled.

Until a couple of releases ago (about 1 year, https://github.com/babel/babel/pull/14538), the default behavior was that generator functions transformed by regenerator relied on a global `regeneratorRuntime` variable. When using `@babel/plugin-transform-runtime` with `regenerator: true` (which is the default), all those global references were replaced with an import to `@babel/runtime/regenerator`.

However now `_regeneratorRuntime` is just a normal Babel helper, with no special handling.

Comment (Andarist) [inline .babelrc.js:15]:
```suggestion
```

**Comment by kossnocorp** [comment]
I've opened a new PR with the commits, this time targeting v2 branch: https://github.com/date-fns/date-fns/pull/3402

Thank you a lot!

**Pull request closed without merging**